### PR TITLE
[1.x] Make `AbstractWorkRegistry` respect the current worker of a node

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
Previously, `ShardProcessingSession` used `NodeId` to indicate who is currently processing which shard. 

On an application node, several workers (threads) can be used to deliver messages. Thus, their identifiers should be used in registry to clearly say who is processing the shard.

This PR updates `AbstractWorkRegistry ` to request the current  node's `WorkerId` to use it in `ShardProcessingSession`. `WorkerId` stores information about the node it belongs and an a worker identifier itself (typically, a thread ID).

The library version is bumped to `1.7.7-SNAPSHOT.8`.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-bin.zip
```
Comment (alexander-yevsyukov) [inline gradle/wrapper/gradle-wrapper.properties:3]: The most recent is 7.3.1 already.

Comment (armiol) [reply gradle/wrapper/gradle-wrapper.properties:3]: @alexander-yevsyukov This is 1.x branch. We cannot use Gradle 7 here.

File: server/src/test/java/io/spine/server/delivery/given/DeliveryTestEnv.java [hunk 2]
```diff
@@ -78,6 +79,14 @@ public static NodeId generateNodeId() {
                 .vBuild();
     }
 
+    public static WorkerId generateWorkerId() {
+        return WorkerId
+                .newBuilder()
+                .setNodeId(generateNodeId())
+                .setValue(String.valueOf(Thread.currentThread().getId()))
```
Comment (armiol) [inline server/src/test/java/io/spine/server/delivery/given/DeliveryTestEnv.java:86]: Please extract the thread ID into the variable.

File: server/src/main/java/io/spine/server/delivery/AbstractWorkRegistry.java [hunk 3]
```diff
@@ -45,50 +45,66 @@
...
+        ShardSessionRecord updatedRecord = updateNode(record, worker);
+        return Optional.of(asSession(updatedRecord));
     }
 
-    private static boolean hasPickedBy(ShardSessionRecord record) {
-        return !NodeId.getDefaultInstance().equals(record.getPickedBy());
+    /**
+     * Returns an identifier of the current worker that is now going to process the shard.
+     *
+     * Typically, it would a thread ID.
```
Comment (armiol) [inline server/src/main/java/io/spine/server/delivery/AbstractWorkRegistry.java:82]: Same as discussed vocally.

Also, in Java we use `<p>`.

**Review by armiol — APPROVED** [review]
@yevhenii-nadtochii LGTM with some things to fix according to the written comments and vocal discussion.

**Pull request merged**